text/0000-app-mixins.md

```diff
@@ -0,0 +1,241 @@
+# Meta
+[meta]: #meta
+- Name: Application Mixins
+- Start Date: 2020-06-22
+- Author(s): [Joe Kutner](https://github.com/jkutner/)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal for allowing application developers (buildpack users) to specify mixins that will be dynamically included with their build and run images.
+
+# Motivation
+[motivation]: #motivation
+
+Mixins already allow buildpack authors to create buildpacks that depend on an extended set of OS packages without affecting build time, but it's common for application code to depend on OS packages too. But allowing buildpacks to arbitrarily install OS packages during build would drastically increase build time, especially when CNB tooling is used to build or rebase many apps with similar package requirements. For that reason, this proposal defines a mechanism to implement mixin support for application developers without significant impact on build performance.
+
+# What it is
+[what-it-is]: #what-it-is
+
+- *application developer* - a person who is using buildpacks to transform their application code into an OCI image
+- *root buildpack* - a new type of buildpack that runs as the root user
+- *stackpack* - a type of root buildpack that runs against the stack image(s) instead of an app
+
+An application developer may specify a list of mixins in their application's `project.toml` file like this:
+
+```toml
+[build]
+mixins = [ "<mixin name>" ]
+```
+
+When a command like `pack build` is run, the list of mixins will be processed before buildpacks are run. For each mixin name, the following will happen:
+
+* If the mixin name is prefixed with `build:` (as per [RFC-0006](https://github.com/buildpacks/rfcs/blob/main/text/0006-stage-specific-mixins.md)) the mixin will be dynamically added to the build image.
+* If the mixin name is prefixed with `run:` (as per [RFC-0006](https://github.com/buildpacks/rfcs/blob/main/text/0006-stage-specific-mixins.md)) the mixin will be dynamically added to the run image.
+* If the mixin name does not have a prefix it will be dynamically added to both the build and run stack images.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+When a list of mixins are defined in a `project.toml` and `pack build` is run against that project, the following will happen:
```

Comment:
I thought they're supposed to come from the build plan (contributed by other buildpacks)? 
```suggestion
When a list of mixins are required by buildpacks via the build plan the build phase starts:
```